controllers/helmrepository_controller.go

```diff
@@ -433,13 +434,15 @@ func (r *HelmRepositoryReconciler) reconcileSource(ctx context.Context, obj *sou
 	chartRepo.Unload()
 
 	// Mark observations about the revision on the object.
-	if !obj.GetArtifact().HasRevision(newChartRepo.Checksum) {
```

Comment:
`newChartRepo.Checksum` is always empty, resulting in this condition to be true always.